### PR TITLE
Add Plugin Priority Settings

### LINKED ISSUE #274: Enable plugin priority
**Describe the solution you'd like**

I'd like plugins to have a settable priority. When there is more than one plugin offering results for an action keyword, the results should be sorted according to that priority.

**Describe alternatives you've considered**

I can't find any.

**Additional context**

I'd like to have the Window Walker plugin to show results before files, links, directories etc. For example, if I <kbd>Alt</kbd>+<kbd>Space</kbd> and write `Edge` and there is already a Microsoft Edge running, it should show before the Microsoft Edge link or executable.

### PR DESCRIPTION
**Pull request by taooceros**
close #274 
This currently only bumps the result score based on their Priority.
After #195 , a better way that won't affect results score can be done by using OrderBy and ThenBy.

### CODE REVIEW

**Comment by taooceros** [comment]
The design of assigning a weight to each plugin is a bit...primitive……. If anyone would have a better design, please feel free to comment. @Flow-Launcher/team

File: Flow.Launcher/PriorityChangeWindow.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+
+namespace Flow.Launcher
+{
+    /// <summary>
+    /// PriorityChangeWindow.xaml 的交互逻辑
```
Comment (jjw24) [inline Flow.Launcher/PriorityChangeWindow.xaml.cs:21]: lets update this auto generated msg to english so it's consistent throughout the project plz

Comment (taooceros) [reply Flow.Launcher/PriorityChangeWindow.xaml.cs:21]: ok

File: Flow.Launcher/SettingWindow.xaml.cs [hunk 2]
```diff
@@ -281,5 +290,6 @@ private void OpenPluginFolder(object sender, RoutedEventArgs e)
         {
             FilesFolders.OpenPath(Path.Combine(DataLocation.DataDirectory(), Constant.Themes));
         }
+
```
Comment (jjw24) [inline Flow.Launcher/SettingWindow.xaml.cs:293]: space?

File: Flow.Launcher/PriorityChangeWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            {
+                pluginViewModel.ChangePriority(newPriority);
+                Close();
+            }
+            else
+            {
+                string msg = "Please provide an valid integer";// translater.GetTranslation("newActionKeywordsHasBeenAssigned");
+                MessageBox.Show(msg);
+            }
+
```
Comment (jjw24) [inline Flow.Launcher/PriorityChangeWindow.xaml.cs:60]: space?

File: Flow.Launcher.Plugin/PluginMetadata.cs [hunk 4]
```diff
@@ -37,11 +37,14 @@ internal set
         public List<string> ActionKeywords { get; set; }
 
         public string IcoPath { get; set;}
-
+        
         public override string ToString()
         {
             return Name;
         }
+        [JsonIgnore]
```
Comment (jjw24) [inline Flow.Launcher.Plugin/PluginMetadata.cs:45]: lets space here please

File: Flow.Launcher/PriorityChangeWindow.xaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                   VerticalAlignment="Center" HorizontalAlignment="Left" />
+
+        <TextBlock FontSize="14" Grid.Row="1" Grid.Column="1" VerticalAlignment="Center"
+                   HorizontalAlignment="Left" Text="New Priority:" />
+        <StackPanel Grid.Row="1" Orientation="Horizontal" Grid.Column="1">
+            <TextBox x:Name="tbAction" Margin="140 10 15 10" Width="105" VerticalAlignment="Center" HorizontalAlignment="Left" />
+        </StackPanel>
+
+        <TextBlock Grid.Row="2" Grid.ColumnSpan="1" Grid.Column="1" Foreground="Gray"
+                   Text="Higher means display more ahead, if you want the results show lower than any other plugins, provide a negative number" TextWrapping="Wrap"/>
```
Comment (jjw24) [inline Flow.Launcher/PriorityChangeWindow.xaml:33]: 1. Could we update the message to:
'Higher the number, the closer the result will be to the top. If you want the results to be lower than any other plugin's, provide a negative number'

2. Also lets add this to the language file so others can translate

3. Lets fix up the width so it doesnt look crowded:

image

**Review by jjw24 — CHANGES_REQUESTED** [review]
got some minor changes from testing

**Review by jjw24 — APPROVED** [review]

**Review by jjw24 — APPROVED** [review]

**Comment by taooceros** [comment]
Should we make the scoring bump higher (50 to 150? ) since sometimes plugins' scores can be low as discussion in #274?

**Pull request merged**